torch/_dynamo/eval_frame.py

```diff
@@ -114,6 +114,54 @@ def _maybe_set_eval_frame(callback: DynamoCallback):
         return set_eval_frame(callback)
 
 
+_phase = "default"
+
+
+def _set_phase(phase: str):
+    global _phase
+    from torch._C._dynamo.eval_frame import set_eval_frame_callback_enabled
+
+    if phase == "default":
+        set_eval_frame_callback_enabled(True)
+    elif phase == "force_eager":
+        set_eval_frame_callback_enabled(False)
+    else:
+        raise AttributeError(f"unsupported torch.compiler phase: {phase}")
+
+    prior = _phase
+    _phase = phase
+    return prior
+
+
+# forbid in graph
+_set_phase._dynamo_forbidden = True  # type: ignore[attr-defined]
+
+
+class SetPhaseCtxMgrDecorator:
```

Comment:
Using manual context manager/decorator class instead of `contextlib.contextmanager` in order to mark a decorated function as forbidden in graph.